js/compiler/compiler.go

```diff
@@ -48,42 +48,58 @@ var (
 	once             sync.Once
 )
 
+// CompatibilityMode specifies the JS compatibility mode
+// nolint:lll
+//go:generate enumer -type=CompatibilityMode -transform=snake -trimprefix CompatibilityMode -output compatibility_mode_gen.go
+type CompatibilityMode uint
+
+// "es6": default, Goja+Babel+core.js
+// "es51": plain Goja
+const (
+	CompatibilityModeES6 CompatibilityMode = iota
+	CompatibilityModeES51
+)
+
 // A Compiler uses Babel to compile ES6 code into something ES5-compatible.
 type Compiler struct {
 	vm *goja.Runtime
 
 	// JS pointers.
-	this      goja.Value
-	transform goja.Callable
-	mutex     sync.Mutex //TODO: cache goja.CompileAST() in an init() function?
+	this              goja.Value
+	transform         goja.Callable
+	mutex             sync.Mutex //TODO: cache goja.CompileAST() in an init() function?
+	compatibilityMode CompatibilityMode
 }
 
-// Constructs a new compiler.
-func New() (*Compiler, error) {
+// New constructs a new compiler with the provided compatibility mode
+func New(compatMode CompatibilityMode) (*Compiler, error) {
 	var err error
 	once.Do(func() {
-		compilerInstance, err = new()
+		compilerInstance, err = new(compatMode)
 	})
 
 	return compilerInstance, err
 }
 
-func new() (*Compiler, error) {
+func new(compatMode CompatibilityMode) (*Compiler, error) {
```

Comment:
Don't name it `new`, it conflicts with Go `new`.

Maybe something like `newCompiler`

Reply:
Can we address this in a new issue/PR?

Reply:
@imiric yes, maybe create an issue first, so we won't forget it.

Reply:
I'm resolving this since I removed `new()` entirely in the most recent refactor. :)

Take a look at `compiler.go` again.